### DISCUSSION TITLE
Self hosted Supabase behind a prefix

### DISCUSSION THREAD

**Discussion by Draela** [supabase/supabase#27248] [Questions]
Hello everyone, 
It's been a few weeks I try to host Supabase on my server behind a Traefik with a prefix "app/supabase/", but every time I try something is not working.

I have seen the repository "https://github.com/supabase-community/supabase-traefik" witch is recommended on the website but this configuration uses subdomain instead of prefix.

I have tried to configure Traefik to remove my prefix before sending the request to Kong, but they are a 301 redirection that rewrites the url and remove my prefix. The new url is not for Traefik one of the Supabase urls then it doesn't follow the request to Supabase (Kong) but to a another service.
For now I use my Supabase behind a subdomain, but I would like to change that.

Did I miss something in the documentation or elsewhere ?
Or is not possible for now to deploy a Supabase self-host stack behind a prefix ?

Thank You.

**Comment by k3rnL** [comment]
As there is no multi-project menu in the supabase self-hosted version, I also wanted to setup multiple instances and make them accessible through a prefix. 
Unfortunately, as of today, this is impossible without rebuilding the docker image and make some modification.
I've spent my day trying to figure out how everything is implemented.

For people who are interested, and have time, there is my conclusion, but it may be more work to do than just this.

1. Instead of doing domain.com/{intance-name}/**, it's easier to simply match you reverse proxy rule to domain.com/project/{instance-name}/** because all the routes accessible in the open source version of supabase studio are under /project/{project}, but the project name does not matter at all. Behind the scene, if the global variable IS_PLATFORM is not set of set at false, the same project is returned whatever is in the path.
2. This lead to a major problem, all the static ressources are served in domain.com/** ... You could bring a supabase instance just to server them, or serve them yourself, but this isn't an option for other routes such as /api ...
3. What would be perfect would be to change the base url isn't it ? Well, this is possible, but it is also impossible. There is this file:

https://github.com/supabase/supabase/blob/master/apps/studio/lib/constants/index.ts
```ts
export const IS_PLATFORM = process.env.NEXT_PUBLIC_IS_PLATFORM === 'true'
export const DEFAULT_HOME = IS_PLATFORM ? '/projects' : '/project/default'
export const API_URL = IS_PLATFORM ? process.env.NEXT_PUBLIC_API_URL : '/api'
export const API_ADMIN_URL = IS_PLATFORM ? process.env.NEXT_PUBLIC_API_ADMIN_URL : undefined
export const PG_META_URL = IS_PLATFORM
  ? process.env.PLATFORM_PG_META_URL
  : process.env.STUDIO_PG_META_URL
export const BASE_PATH = process.env.NEXT_PUBLIC_BASE_PATH ?? ''
```
Cool, but also not cool. Because this is a nextjs app, and those variables are set at compile time (I can be wrong here, I don't work with JS/TS at all). And in https://github.com/supabase/supabase/blob/master/apps/studio/next.config.js there is this part : 
```js
const nextConfig = {
  basePath: process.env.NEXT_PUBLIC_BASE_PATH,
  output: 'standalone',
.......
```
And this variable is set at compile time, at the end, a server.js file is compiled and if we look at it in the docker container, the baseUrl is empty. 

A quick fix would be to mount a custom server.js that changes this variable directly in the docker container, or it's possible to rebuild the container with you own variable. 
But, if you look at the constants above, the baseUrl will effectively affect all the routes that are served, but will not changes how the routes are USED. For this, you would need to make IS_PLATFORM=true. But will certainly lead to many problem, it's not an option. And I guess the platform source code isn't public, as it's not on GitHub. So here we need for sure to modify the code.
Also, in many places in the code, they take the public url, and extract the domain. 

I'm 80% sure that those modification would make anyone able to create an instance behind a prefixed url. 

Btw, I'm sure you noticed that some menu are not present in the self-hosted version. It's all because of this "IS_PLATFORM". Make it true and you have the menus, but you will have to implement you are platform and serve all the routes /platform/** that they call in the studio.
